src/Apps/Artwork/Components/ArtworkDetails/RequestConditionReport.tsx

```diff
@@ -0,0 +1,205 @@
+          resolve(data)
+        },
+        onError: error => {
+          reject(error)
+        },
+        mutation: graphql`
+          mutation RequestConditionReportMutation(
+            $input: RequestConditionReportInput!
+          ) {
+            requestConditionReport(input: $input) {
+              conditionReportRequest {
+                internalID
+              }
+            }
+          }
+        `,
+        variables: {
+          input: { saleArtworkID },
+        },
+      })
+    })
+  }
+
+  const handleMutationError = error => {
+    logger.error(error)
+
+    let errorMessages: string[]
+    if (Array.isArray(error)) {
+      errorMessages = error.map(e => e.message)
+    } else if (typeof error === "string") {
+      errorMessages = [error]
+    } else if (error.message) {
+      errorMessages = [error.message]
+    }
+
+    setRequestingConditionReport(false)
+    trackRequestConditionReportFailure(errorMessages)
+    setShowErrorModal(true)
+  }
+
+  const commonProperties = {
+    sale_artwork_id: saleArtworkID,
+    user_id: user && user.id,
+  }
+
+  function trackRequestConditionReportFailure(errors: string[]) {
+    trackEvent({
+      action_type: Schema.ActionType.RequestConditionReportFailed,
+      error_messages: errors,
+      ...commonProperties,
+    })
+  }
+
+  function trackRequestConditionReportSuccess() {
+    trackEvent({
+      action_type: Schema.ActionType.RequestConditionReportSubmitted,
+      ...commonProperties,
+    })
+  }
+
+  const handleLoginClick = () => {
+    trackEvent({
+      action_type: AnalyticsSchema.ActionType.Click,
+      subject: AnalyticsSchema.Subject.Login,
+    })
+
+    return authentication.login(mediator)
+  }
+
+  const handleRequestConditionReportClick = () => {
+    setRequestingConditionReport(true)
+
+    requestConditionReport()
+      .then(data => {
+        const theData = data as any
+        if (theData.requestConditionReport) {
+          setRequestingConditionReport(false)
+          trackRequestConditionReportSuccess()
+          setShowConditionReportRequestedModal(true)
+        } else {
+          handleMutationError("Unknown error")
+        }
+      })
+      .catch(error => {
+        handleMutationError(error)
+      })
+  }
+
+  const UnauthenticatedContent: React.FC = () => (
+    <>
+      <Button size="small" variant="secondaryGray" onClick={handleLoginClick}>
+        Log in
+      </Button>
+      <Sans display="inline" ml={1} size="2">
+        to request
+      </Sans>
+    </>
+  )
+
+  const AuthenticatedContent: React.FC = () => (
+    <Button
+      size="small"
+      width={180}
+      variant="secondaryGray"
+      onClick={handleRequestConditionReportClick}
+    >
+      {requestingConditionReport ? (
+        <Spinner size="small" />
+      ) : (
+        "Request condition report"
+      )}
+    </Button>
+  )
+
+  const RequestedConditionReportModal: React.FC = () => (
+    <Modal
+      title="Condition report requested"
```

Comment:
Do we have a good sense of all the different places a modal can surface on this page? I'm thinking of registration with checkbox, confirmation, login, this... I wish there were a single interface to manage all these though I guess that is out of scope of this app (and I guess it still wouldn't on its own answer the question of what happens if multiple components try to trigger a modal)

Reply:
The login flow is the only modal triggered by the `mediator` event hub here. I don't think the other modals would conflict. The event handling should be isolated enough not to cause issues. I actually prefer that isolation rather than having a global state of modals.